trio/_ssl.py

```diff
@@ -239,9 +239,12 @@ def done(self) -> bool:
 
 _State = _Enum("_State", ["OK", "BROKEN", "CLOSED"])
 
+# TODO: variance
+T_Stream = TypeVar("T_Stream", bound=Stream)
```

Comment:
I think this does need to be invariant. For `SSLStream` it could be covariant since we never use it as a return value, but `SSLListener` does return it. Could be covariant for the former and invariant for the latter, but that's just confusing.